collector/src/main.rs

```diff
@@ -745,137 +686,196 @@ fn main() {
+        #[clap(long, default_value = "1")]
+        iterations: usize,
 
-        (@subcommand diff_local =>
-            (about: "Profiles and compares two toolchains with one of several profilers")
-
-            // Mandatory arguments
-            (@arg PROFILER: +required +takes_value
-             "One of: 'self-profile', 'time-passes', 'perf-record',\n\
-             'oprofile', 'cachegrind', 'callgrind', 'dhat', 'massif',\n\
-             'eprintln', 'llvm-lines', 'mono-items', 'dep-graph'")
-            (@arg RUSTC_BEFORE: +required +takes_value "The path to the local rustc to benchmark")
-            (@arg RUSTC_AFTER:  +required +takes_value "The path to the local rustc to benchmark")
-
-             // Options
-            (@arg BUILDS: --builds       +takes_value
-             "One or more (comma-separated) of: 'Check', \n\
-             'Debug', 'Doc', 'Opt', 'All'")
-            (@arg CARGO:   --cargo       +takes_value "The path to the local Cargo to use")
-            (@arg EXCLUDE: --exclude     +takes_value
-             "Exclude all benchmarks that are listed in\n\
-             this comma-separated list of patterns")
-            (@arg INCLUDE: --include     +takes_value
-             "Include only benchmarks that are listed in\n\
-             this comma-separated list of patterns")
-            (@arg OUT_DIR: --("out-dir") +takes_value "Output directory")
-            (@arg RUNS:    --runs        +takes_value
-             "One or more (comma-separated) of: 'Full',\n\
-             'IncrFull', 'IncrUnchanged', 'IncrPatched', 'All'")
-            (@arg RUSTDOC: --rustdoc +takes_value "The path to the local rustdoc to benchmark")
-        )
+        #[clap(flatten)]
+        self_profile: SelfProfileOption,
+    },
 
-        (@subcommand install_next =>
-            (about: "Installs the next commit for perf.rust-lang.org")
+    /// Benchmarks the next commit for perf.rust-lang.org
+    BenchNext {
+        /// Site URL
+        site_url: String,
 
-            // Mandatory arguments: (none)
+        #[clap(flatten)]
+        db: DbOption,
 
-            // Options: (none)
-        )
-    )
-    .get_matches();
+        #[clap(flatten)]
+        bench_rustc: BenchRustcOption,
+
+        #[clap(flatten)]
+        self_profile: SelfProfileOption,
+    },
+
+    /// Benchmarks a published toolchain for perf.rust-lang.org's dashboard
+    BenchPublished {
+        /// Toolchain (e.g. stable, beta, 1.26.0)
+        toolchain: String,
+
+        #[clap(flatten)]
+        db: DbOption,
+    },
+
+    /// Profiles a local rustc with one of several profilers
+    ProfileLocal {
+        /// Profiler to use
+        #[clap(arg_enum)]
+        profiler: Profiler,
+
+        #[clap(flatten)]
+        rustc: RustcOption,
+
+        #[clap(flatten)]
+        id: IdOption,
+
+        #[clap(flatten)]
+        local: LocalOptions,
+
+        /// Output directory
+        #[clap(long = "out-dir", default_value = "results/")]
```

Comment:
ditto about inferred `long`

Reply:
This is the case where it ends up as `--out_dir` if I don't have the explicit name.